agent/internal/agent.go

```diff
@@ -245,18 +245,37 @@ func (a *agent) setup(ctx *actor.Context) error {
 	ctx.Log().Infof("Determined agent %s (built with %s)", a.Version, runtime.Version())
 	actors.NotifyOnSignal(ctx, syscall.SIGINT, syscall.SIGTERM)
 
-	if a.ArtificialSlots > 0 {
+	switch {
```

Comment:
non-blocking: can we pull this whole switch into a func that takes slot type/artificial slots/etc and returns devices and an error and move it back to detect.go? Trying to avoid agent.go becoming like trial.go 🤣